src/services/analytics/gtm.ts

```diff
@@ -0,0 +1,90 @@
+import TagManager, { TagManagerArgs } from 'react-gtm-module'
+import Cookies from 'js-cookie'
+import {
+  IS_PRODUCTION,
+  GOOGLE_TAG_MANAGER_ID,
+  GOOGLE_TAG_MANAGER_AUTH_LIVE,
+  GOOGLE_TAG_MANAGER_AUTH_LATEST,
+  GOOGLE_TAG_MANAGER_DEVELOPMENT_AUTH,
+} from '@/config/constants'
+import { AnalyticsEvent, EventLabel, EventType } from './types'
+
+type GTMEnvironment = 'LIVE' | 'LATEST' | 'DEVELOPMENT'
+type GTMEnvironmentArgs = Required<Pick<TagManagerArgs, 'auth' | 'preview'>>
+
+const GOOGLE_ANALYTICS_COOKIE_LIST = ['_ga', '_gat', '_gid']
+
+const GTM_ENV_AUTH: Record<GTMEnvironment, GTMEnvironmentArgs> = {
+  LIVE: {
+    auth: GOOGLE_TAG_MANAGER_AUTH_LIVE,
+    preview: 'env-1',
+  },
+  LATEST: {
+    auth: GOOGLE_TAG_MANAGER_AUTH_LATEST,
+    preview: 'env-2',
+  },
+  DEVELOPMENT: {
+    auth: GOOGLE_TAG_MANAGER_DEVELOPMENT_AUTH,
+    preview: 'env-3',
+  },
+}
+
+let _chainId: string = ''
+
+export const setChainId = (chainId: string): void => {
+  _chainId = chainId
+}
+
+export const gtmInit = (): void => {
+  const GTM_ENVIRONMENT = IS_PRODUCTION ? GTM_ENV_AUTH.LIVE : GTM_ENV_AUTH.DEVELOPMENT
+
+  if (!GOOGLE_TAG_MANAGER_ID || !GTM_ENVIRONMENT.auth) {
+    console.warn('[GTM] - Unable to initialize Google Tag Manager. `id` or `gtm_auth` missing.')
+    return
+  }
+
+  TagManager.initialize({
+    gtmId: GOOGLE_TAG_MANAGER_ID,
+    ...GTM_ENVIRONMENT,
+    dataLayer: {
+      // Block JS variables and custom scripts
```

Comment:
Is there a reason we are not adding `event: GTM_EVENT.PAGEVIEW` etc. here anymore? If I remember correctly we used that custom event to trigger an initial page view

Reply:
I'm planning to trigger pageViews from another place. This fn here will just intialize GTM.